### PR TITLE
Issue 194 sso using google oauth

### LINKED ISSUE #194: SSO using google oauth2

### PR DESCRIPTION
**Pull request by iffyuva**
Fixes #194

### CODE REVIEW

**Comment by emilsoman** [comment]
@iffyuva actually this was already implemented on a different branch : https://github.com/code-mancers/dockup/commit/a005ca8a93f5fef43c7315a641799b9364e1f305#diff-56e5712002c9206b668ee6638175cac1 . See our old dockup URL which has google auth 🙂

File: apps/dockup_ui/web/controllers/auth_controller.ex [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# https://github.com/ueberauth/ueberauth_example
+defmodule DockupUi.AuthController do
+  use DockupUi.Web, :controller
+  plug Ueberauth
+
+  alias Ueberauth.Strategy.Helpers
+
+  def request(conn, _params) do
+    render(conn, "request.html", callback_url: Helpers.callback_url(conn))
+  end
```
Comment (emilsoman) [inline apps/dockup_ui/web/controllers/auth_controller.ex]: Request phase needs to be implemented only if we are presenting our form, in the cause of google oauth, we can let ueberauth do the default thing of redirecting to google. (I did not implement this in my branch it was working as expected)

File: apps/dockup_ui/web/router.ex [hunk 2]
```diff
@@ -14,14 +16,23 @@ defmodule DockupUi.Router do
...
     resources "/whitelisted_urls", WhitelistedUrlController, except: [:index, :show]
   end
 
+  scope "/auth", DockupUi do
+    pipe_through :browser
+
+    get "/:provider", AuthController, :request
+    get "/:provider/callback", AuthController, :callback
+
+    resources "/", AuthController, singleton: :true
```
Comment (emilsoman) [inline apps/dockup_ui/web/router.ex]: Can we specify the actions that we need here using `only` ?

File: apps/dockup_ui/lib/dockup_ui/config.ex [hunk 3]
```diff
@@ -13,7 +13,10 @@ defmodule DockupUi.Config do
       {"SLACK_WEBHOOK_URL", :slack_webhook_url, :string},
       {"DOCKUP_BACKEND", :backend_module, :module},
       {"DOCKUP_HIBERNATE_ALL_AT", :hibernate_all_at, :string},
-      {"DOCKUP_WAKEUP_ALL_AT", :wakeup_all_at, :string}
+      {"DOCKUP_WAKEUP_ALL_AT", :wakeup_all_at, :string},
+      {"DOCKUP_GOOGLE_CLIENT_ID", :google_client_id, :string},
+      {"DOCKUP_GOOGLE_CLIENT_SECRET", :google_client_secret, :string},
+      {"DOCKUP_GOOGLE_DOMAINS", :google_client_domains, :array},
```
Comment (emilsoman) [inline apps/dockup_ui/lib/dockup_ui/config.ex]: Can you link the related PR for orchestration that adds these values too please in this PR's description?

File: apps/dockup_ui/config/config.exs [hunk 4]
```diff
@@ -32,3 +32,9 @@ config :phoenix, :generators,
   binary_id: false
 
 config :dockup_ui, ecto_repos: [DockupUi.Repo]
+
+# Configure google oauth
+config :ueberauth, Ueberauth,
+  providers: [
+    google: {Ueberauth.Strategy.Google, [default_scope: "email profile"]}
```
Comment (emilsoman) [inline apps/dockup_ui/config/config.exs]: Do we need these scopes now? I guess we are just letting people pass through the auth and don't need their info for anything right?

**Comment by iffyuva** [comment]
I didn't realize that, let me re-use that commit :)

**Pull request closed without merging**